### ISSUE TITLE
[BUG] Inconsistent parsing of identifiers with PowerBI

### ISSUE THREAD

**Issue by UliPlabst** [microsoft/powerquery-parser#269]
A user of our powerquery formatter service just brought iconsistent behavior to our attention. The relevant issue in our repo is [here](https://github.com/mogulargmbh/m-formatter/issues/12).

**Expected behavior**
Consistent parsing behavior of identifiers of powerquery-parser and the internal m-parser of PowerBI 

**Actual behavior**
PowerBI will allow parsing of identifiers such as `MyIdentifier.1` while this library fails. Looking at the powerquery language spec
```
identifier:
      regular-identifier
      quoted-identifier
regular-identifier:
      available-identifier
      available-identifier dot-character regular-identifier
available-identifier:
      A keyword-or-identifier that is not a keyword
keyword-or-identifier:
      letter-character
      underscore-character
      identifier-start-character identifier-part-characters
identifier-start-character:
      letter-character
      underscore-character
identifier-part-characters:
      identifier-part-character identifier-part-charactersopt
identifier-part-character:
      letter-character
      decimal-digit-character
      underscore-character
      connecting-character
      combining-character
      formatting-character
```
the Identifier `MyIdentifier.1` is actually an invalid identifier. So your library is consistent with the spec but power bi isn't. This isn't really a critical bug but I still wanted to file it with you in case you want to address it. I filed it with you instead of over at the powerbi forum because it will just get buried over there.

**To Reproduce**
Please include the following:
* (Required) The Power Query script that triggers the issue.
`let MyIdentifier.1 in MyIdentifier.1`
* (Required) Any non-default settings used in the API call(s) which trigger the issue.
none
* (Ideally) A minimal reproducible example. Can you reproduce the problem by calling a function in `src/example.ts`?

**Additional context**

**Comment by JordanBoltonMN** [comment]
You are correct! The ultimate goal of this parser is to be consistent with the original parser, even if that doesn't follow the official spec. I was hoping to get around to this sooner rather than later, but I might wait until IE 11 support support is dropped. That should land soon-ish. The reason being IE 11 has some really annoying Regexp Unicode issues and if I'm going to take a pass on it I'd prefer it being only once :)

**Comment by ninmonkey** [comment]
Maybe the same issue as: https://github.com/microsoft/vscode-powerquery/issues/93

# Environment




| Name                      |               Version                                         |
| ---------------------------- | ------------------------------------------------------ |
| Power BI Desktop | Version: 2.103.881.0 64-bit (March 2022) |
| code                         | 1.66.0, e18005f0f1b33c29e81d732535d8c0e47cafb0b5 , x64 |
| PowerQuery.vscode-powerquery | 0.1.29                                                 |

# Example Test Cases

<table>


<tr>
<td>Code</td>
<td>Power BI</td>
<td>VS Code</td>
</tr>
<tr>
<td>
<pre>
let     
    bar.1 = 2
in
    bar.1
</pre>
</td>
<td>✅ returns <code>2</code></td>
<td>⛔ Expected to find a equal operator <code><'='></code>, but a numeric literal was found
            instead<br />powerquery(Error.Parse)</td>
</tr>

<tr>
<td>Code</td>
<td>Power BI</td>
<td>VS Code</td>
<tr>
<td>
<pre>
let     
    #"bar.1" = 2,
    foo = #"bar.1"
in
    foo
</pre>
</td>
<td>✅ returns <code>1</code></td>
<td>✅ returns <code>[let-variable] foo: 2</code></td>
</tr>

<tr>
<td>Code</td>
<td>Power BI</td>
<td>VS Code</td>
<tr>
<td>
<pre>
let   
    y = a.
in
    y
</pre>
</td>
<td>⛔ invalid identifier</td>
<td>? No error, but no info on tooltip (not even any)</td>
</tr>



</table>

**Comment by JordanBoltonMN** [comment]
Should be resolved with #301

**Comment by UliPlabst** [comment]
Cool thanks
